xarray/core/accessors.py

```diff
@@ -58,6 +58,49 @@ def _get_date_field(values, name, dtype):
         return _access_through_series(values, name)
 
 
+def _round_series(values, name, freq):
+    """Coerce an array of datetime-like values to a pandas Series and
+    apply requested rounding
+    """
+    values_as_series = pd.Series(values.ravel())
+    if name == 'floor':
+        field_values = values_as_series.dt.floor(freq=freq).values
```

Comment:
can you rewrite this to avoid the conditionals? maybe something like:
```python
method = getattr(values_as_series.dt, name)
field_values = method(freq=freq).values
```

Reply:
Yes! I didn't know how to do this. The above works.